examples/ca/hpilo_ca

```diff
@@ -133,11 +133,19 @@ def sign_certificates(hosts, openssl, login, password, opts):
                 cn = ilo.get_cert_subject_info()['csr_subject_common_name']
 
         hn, dn = hostname.split('.', 1)
-        if network_settings['dns_name'] != hn or dn and network_settings['domain_name'] != dn:
-           print("Hostname misconfigured on the ilo, fixing")
-           ilo.mod_network_settings(dns_name=hn, domain_name=dn)
-           todo.append(hostname)
-           continue
+    	if network_settings["dhcp_domain_name"]:
+    		# Note: Domain Name is currently set through DHCP, if enabled then domain_name is empty and hpilo_ca can't update. Avoiding loop.
+    		if network_settings['dns_name'] != hn:
+                print("Hostname misconfigured on the ilo - hn:[%s] dn:[%s], fixing") % (network_settings['dns_name'], hn)
+                ilo.mod_network_settings(dns_name=hn, domain_name=dn)
```

Comment:
This is still trying to update the domain name, isn't that incorrect?

Reply:
It will update the domain name if iLO is not set to use DHCP. The loop happens when iLO is set to use DHCP. See attached screenshot for better understanding.

![screen shot 2018-08-23 at 2 40 35 pm](https://user-images.githubusercontent.com/801289/44553535-a2443980-a6e2-11e8-8a01-759b98a1f6b0.png)